File: sample/src/main/java/com/raxdenstudios/square/sample/AutoInflateLayoutActivity.java
package com.raxdenstudios.square.sample;

import android.os.Bundle;
import android.view.View;

import com.raxdenstudios.square.SquareActivity;
import com.raxdenstudios.square.interceptor.interactor.AutoInflateLayoutInterceptorInteractor;
import com.raxdenstudios.square.interceptor.type.activity.AutoInflateLayoutActivityInterceptor;

public class AutoInflateLayoutActivity extends SquareActivity implements AutoInflateLayoutActivityInterceptor {

    AutoInflateLayoutInterceptorInteractor mAutoInflateLayoutInteractor;

    private View mView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onContentViewCreated(View view, Bundle savedInstanceState) {
        mView = view;
    }

    @Override
    public void onInterceptorAttached(AutoInflateLayoutInterceptorInteractor interceptor) {
        mAutoInflateLayoutInteractor = interceptor;
    }


}

File: sample/src/main/java/com/raxdenstudios/square/sample/commons/AutoInflateLayoutActivity.kt
package com.raxdenstudios.square.sample.commons

import android.support.v7.app.AppCompatActivity
import android.view.View
import com.raxdenstudios.square.interceptor.Interceptor
import com.raxdenstudios.square.interceptor.commons.autoinflatelayout.AutoInflateLayoutInterceptor
import com.raxdenstudios.square.interceptor.commons.autoinflatelayout.HasAutoInflateLayoutInterceptor

class AutoInflateLayoutActivity : AppCompatActivity(),
        HasAutoInflateLayoutInterceptor {

    private var mAutoInflateLayoutInterceptor: AutoInflateLayoutInterceptor? = null

    var mContentView: View? = null

    // ======== HasInflateLayoutInterceptor ====================================================

    override fun onContentViewCreated(view: View) {
        mContentView = view
    }

    override fun onInterceptorCreated(interceptor: Interceptor) {
        mAutoInflateLayoutInterceptor = interceptor as? AutoInflateLayoutInterceptor
    }
}